### PR TITLE
fix shuttle calldown memes

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
this was some dumb code but not mine for once

## Changelog
:cl:
fix: fixes shuttle calldown memes
/:cl:

### CODE REVIEW

File: code/modules/shuttle/shuttle.dm [hunk 1]
```diff
@@ -420,6 +420,7 @@
 			mode = SHUTTLE_IGNITING
 			on_ignition()
 			setTimer(ignitionTime)
+			SEND_SIGNAL(src, COMSIG_SHUTTLE_IGNITING, destination)
```
Comment (Rohesie) [inline code/modules/shuttle/shuttle.dm]: This should be before `on_ignition()`

File: code/modules/shuttle/shuttle.dm [hunk 2]
```diff
@@ -579,6 +580,7 @@
 				mode = SHUTTLE_PREARRIVAL
 				on_prearrival()
 				setTimer(prearrivalTime)
+				SEND_SIGNAL(src, COMSIG_SHUTTLE_PREARRIVAL, destination)
```
Comment (Rohesie) [inline code/modules/shuttle/shuttle.dm]: Same here for `on_prearrival()`

File: code/modules/shuttle/shuttle.dm [hunk 3]
```diff
@@ -606,11 +610,13 @@
 				mode = SHUTTLE_CALL
 				setTimer(callTime * engine_coeff)
 				enterTransit()
+				SEND_SIGNAL(src, COMSIG_SHUTTLE_CALL, destination)
```
Comment (Rohesie) [inline code/modules/shuttle/shuttle.dm]: And this for `enterTransit()`

File: code/modules/shuttle/shuttle.dm [hunk 4]
```diff
@@ -606,11 +610,13 @@
...
 				return
 
 	mode = SHUTTLE_IDLE
 	timer = 0
 	destination = null
+	SEND_SIGNAL(src, COMSIG_SHUTTLE_IDLE)
```
Comment (Rohesie) [inline code/modules/shuttle/shuttle.dm]: Having this on a `set_shuttle_idle()` or `set_shuttle_status(SHUTTLE_IDLE)` proc might be better.

File: code/modules/shuttle/shuttle.dm (hunk 1 above)
Comment (spookydonut) [reply code/modules/shuttle/shuttle.dm]: no it shouldnt

File: code/modules/shuttle/shuttle.dm (hunk 2 above)
Comment (spookydonut) [reply code/modules/shuttle/shuttle.dm]: nope.

File: code/modules/shuttle/shuttle.dm (hunk 3 above)
Comment (spookydonut) [reply code/modules/shuttle/shuttle.dm]: nope

File: code/modules/shuttle/shuttle.dm (hunk 4 above)
Comment (spookydonut) [reply code/modules/shuttle/shuttle.dm]: No that would cause infinite recursion if anything registered to this signal called those procs

**Comment by Henriquekill9576-taken** [comment]
if you call the shuttle while its still on the air now it will land on the ship and you wont be able to call it anymore

**Comment by Rohesie** [comment]
Players report the second hijack attempt of the queen failed, bugging the shuttle. I took a picture and it was landed on the ship, showing that.

H13MNxp

**Comment by Eskjjlj** [comment]
https://github.com/tgstation/TerraGov-Marine-Corps/issues/3189 issue about the shuttle transit shutters remaining closed upon landing.

**Comment by Metroidlover** [comment]
Somehow this PR managed to kill the server on initialization.

**Comment by spookydonut** [comment]
> 
> 
> Somehow this PR managed to kill the server on initialization.

Not possible.

**Comment by Rohesie** [comment]
This has already been fixed.

**Pull request closed without merging**